pkg/orchestration/wiring/k8scompute/plugin.go

```diff
@@ -281,6 +287,26 @@ func WireUp(resource *orch_v1.StateResource, context *wiringplugin.WiringContext
 		Path:     metadataNamePath,
 	}
 
+	// Add pod disruption budget
+	pdbSpec := buildPodDisruptionBudgetSpec(resource.Name)
+	pdb := smith_v1.Resource{
+		Name:       wiringutil.ResourceNameWithPostfix(resource.Name, pdbPostfix),
+		References: []smith_v1.Reference{deploymentNameRef},
```

Comment:
Why is this reference here? Nothing is needed from Deployment for PDB, right? We don't use a reference for the labels because they are in a different format, but also the order of creation should probably be reversed. PDB should be created first and then the Deployment. Deployment may take time to finish rolling out. But while this is happening we still want to have the PDB constraints be there. Otherwise all pods may end up on a single host =)

Reply:
The affinity config on the pod handles the spread, the pod disruption budget is simply meant to  prevent us nuking all pods eg. some sort of incident + an upgrade causing too much churn.

Reply:
Reference has been removed.

Reply:
Sorry, confused two things. So the PDB should either be created before the deployment or independently of it, all good.